compiler/qsc_codegen/src/qir_base.rs

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use num_bigint::BigUint;
+use num_complex::Complex;
+use qsc_data_structures::index_map::IndexMap;
+use qsc_eval::{
+    backend::Backend,
+    debug::{map_hir_package_to_fir, Frame},
+    eval_expr,
+    output::GenericReceiver,
+    val::{self, GlobalId, Value},
+    Env, Error, Global, NodeLookup, State,
+};
+use qsc_fir::fir::{BlockId, ExprId, ItemKind, PackageId, PatId, StmtId};
+use qsc_frontend::compile::PackageStore;
+use qsc_hir::hir::{self};
+use quantum_sparse_sim::QuantumSim;
+use std::fmt::Write;
+
+const PREFIX: &str = include_str!("./qir_base/prefix.ll");
+const POSTFIX: &str = include_str!("./qir_base/postfix.ll");
+
+/// # Errors
+///
+/// This function will return an error if execution was unable to complete.
+pub fn generate_qir(
+    store: &PackageStore,
+    package: hir::PackageId,
+) -> Result<String, (Error, Vec<Frame>)> {
+    let mut fir_lowerer = qsc_eval::lower::Lowerer::new();
+    let mut fir_store = IndexMap::new();
+    let package = map_hir_package_to_fir(package);
+    let mut sim = BaseProfSim::default();
+    write!(&mut sim.instrs, "{PREFIX}").expect("writing to string should succeed");
```

Comment:
This file has a lot of `write!(string, <string literal>).expect(...)`. Why not:

```suggestion
    sim.instrs.push_str(PREFIX);
```

Reply:
Updated to usages of `push_str` and `write_fmt`.